src/main/java/com/ita/if103java/ims/service/impl/WarehouseServiceImpl.java

```diff
@@ -43,21 +60,32 @@ public WarehouseDto add(WarehouseDto warehouseDto) {
         Integer maxWarehouseDepth = userDetails.getAccountType().getMaxWarehouseDepth();
         int parentLevel = warehouseDao.findLevelByParentID(warehouseDto.getParentID());
         if (parentLevel + 1 < maxWarehouseDepth) {
-            return createNewWarehouse(warehouseDto);
+            return createNewWarehouse(warehouseDto,userDetails);
         } else {
             throw new MaxWarehouseDepthLimitReachedException("The maximum depth of warehouse's levels has been reached for this" +
                 "{accountId = " + accountId + "}");
         }
     }
 
-    private WarehouseDto createNewWarehouse(WarehouseDto warehouseDto) {
-        Warehouse warehouse = warehouseDtoMapper.toEntity(warehouseDto);
+    private WarehouseDto createNewWarehouse(WarehouseDto warehouseDto, UserDetailsImpl user) {
+        Warehouse warehouse = warehouseDao.create(warehouseDtoMapper.toEntity(warehouseDto));
+        Address address = addressDtoMapper.toEntity(warehouseDto.getAddressDto());
+        addressDao.createWarehouseAddress(warehouse.getId(),address);
+        EventName eventName = EventName.WAREHOUSE_CREATED;
+        createEvent(user, warehouse, eventName);
```

Comment:
Can be simplified 
`createEvent(user, warehouse, EventName.WAREHOUSE_CREATED);`

Please add event message "Created Warehouse '{name}' with level '{level}' <if level != 0 then "as a child of warehouse id '{parent_id}'">"
For example:
Created Warehouse 'Storage' with level '0'
Created Warehouse 'Section A' with level '1' as a child of warehouse id '5'